Update in-repo app to include icon list

MelSumner opened the pull request:
If merged, this PR adds a link to the basic usage instructions and a rendered icon list to the in-repo app. 

![image](https://user-images.githubusercontent.com/4587451/130241665-ac11b237-1337-4ffe-81f5-9f51a587e67d.png)

hashicorp-cla commented:
[![CLA assistant check](https://cla.hashicorp.com/pull/badge/not_signed)](https://cla.hashicorp.com/hashicorp/flight?pullRequest=63)


Thank you for your submission! We require that all contributors sign our Contributor License Agreement ("CLA") before we can accept the contribution. [Read and sign the agreement](https://cla.hashicorp.com/hashicorp/flight?pullRequest=63)

[Learn more about why HashiCorp requires a CLA and what the CLA includes](https://www.hashicorp.com/cla)

<sub>Have you signed the CLA already but the status is still pending? [Recheck it](https://cla.hashicorp.com/check/hashicorp/flight?pullRequest=63).</sub>

Dhaulagiri commented:
> right now it feels really tedious

It _should_ be feasible for us to automatically generate each icon based on some array of inputs (either supplied as part of the icon export or we could reach into the sprite and pull them out I think)

amyrlam reviewed (approved):
pulled it locally, lgtm! 

it wasn't clear to me that we were doing a non-dynamic list as a first pass until the meeting, which is why i hadn't reviewed it before. 

it would be cool to have the instructions and https://github.com/hashicorp/flight/blob/main/ember-flight-icons/README.md be one source of truth in the future

amyrlam commented:
p.s. not sure why CI is stalled yet. i don't had admin privileges to override and merge. will investigate...

amyrlam commented:
my guess is this needs a rebase cause it doesn't show up in https://github.com/hashicorp/flight/actions if you can try that

MelSumner commented:
> It _should_ be feasible for us to automatically generate each icon based on some array of inputs (either supplied as part of the icon export or we could reach into the sprite and pull them out I think)

@Dhaulagiri yup! 

@amyrlam and I paired on this today and it _almost_ works. We pulled the `Name` from the `_catalog.json` file, and used a helper to remove the `-##` from the end of the name. This still renders two of each icon, but just in the default size. So need to think through it some more. 

![image](https://user-images.githubusercontent.com/4587451/129983047-8dd2cc33-396f-487f-9d80-7f72675c228d.png)

MelSumner commented:
Ok, figured everything out. This PR is ready to review @hashicorp/design-systems

Dhaulagiri commented on ember-flight-icons/tests/dummy/app/routes/application.js:
```diff
@@ -0,0 +1,27 @@
+    // console.log('here', this.contextRootURL)
+
+    const response = await fetch(
+      `${this.contextRootURL}@hashicorp/ember-flight-icons/icons/_catalog.json`
```
thought: how will we ensure that this file gets updated?

amyrlam replied:
yeah good one

(non-blocking) want to add, can you in a separate PR consider documenting the https://github.com/hashicorp/flight/tree/main/export-go/dist to https://github.com/hashicorp/flight/tree/main/ember-flight-icons/public/icons workflow, perhaps https://github.com/hashicorp/flight? 

alternatively... i think ideally we just want one source of truth. perhaps we can just use `export-go/dist` as `@hashicorp/flight-icons` ... then have `@hashicorp/flight-icons-ember` (to be renamed, if we agree) consume `@hashicorp/flight-icons`

i'm going to muddle with `@hashicorp/flight-icons` and make another version to npm, but i might try making the file from `export-go/dist` instead today

Dhaulagiri replied:
right now it's pretty ad hoc (which is ok!) but we need to map out how exactly we want to connect the dots here, and I agree having a source of truth will be better than relying on manual copies. having the file live with the contents of `hashicorp/flight-icons` seems reasonable to me and is similar to what we do with structure-icons currently. For the purposes of documentation generation it _is_ possible to read the contents of the sprite itself and generate the list of icons based off of that, but that's sort of the same difference as reading from that file I suppose.

Dhaulagiri commented on ember-flight-icons/tests/dummy/app/routes/application.js:
```diff
@@ -0,0 +1,27 @@
+  }
+
+  async model() {
+    // console.log('here', this.contextRootURL)
```
nitpick (non-blocking): could remove this if it's no longer needed

Dhaulagiri commented on ember-flight-icons/tests/dummy/app/helpers/substring.js:
```diff
@@ -0,0 +1,8 @@
+import Helper from '@ember/component/helper';
+
+export default class Substring extends Helper {
```
suggestion: add a comment about the intended behavior of this helper

MelSumner replied:
ok, added 👍

Dhaulagiri commented on ember-flight-icons/tests/integration/helpers/substring-test.js:
```diff
@@ -0,0 +1,17 @@
+  setupRenderingTest(hooks);
+
+  // TODO: Replace this with your real tests.
+  test('it renders', async function (assert) {
```
nitpick (non-blocking): if we don't think a real test is necessary we could delete this test file

MelSumner replied:
agree; removed.

Dhaulagiri commented on ember-flight-icons/app/helpers/substring.js:
```diff
@@ -0,0 +1 @@
+export { default } from '@hashicorp/ember-flight-icons/helpers/substring';
```
question: do we want to expose this helper to consumers of the addon?

MelSumner replied:
no, they don't need it. we won't (eventually) need it either.

Dhaulagiri replied:
not a blocker because I may be mis-remembering, but isn't there a way to create a helper just for the dummy app?

Dhaulagiri reviewed (approved):
👏 

Had a few non-blocking points of feedback/questions to consider before merging

amyrlam commented:
👍 looks good, nice one!

separate from this PR, i'd love for us to discuss WIP commits and github stuff, maybe at retro? it was a bit hard to review commit-by-commit. i tend to squash out WIP commits. would love to come to an agreement as a team. 
- in this PR specifically, it would be cool if adding `ember-fetch` and the corresponding package.json / yarn.lock changes as a separate commit so that it's easier to review
- i prob would have dropped the initial commit with all the non-dynamic makeup as well. but that could just be my opinion!
- i've been on teams where everyone uses https://gitmoji.dev/ to keep commits as atomic as possible 

alternatively, we could consider squashing PR's on merge (can be challenging looking back at the history if the PR's are super large, but can always go back to the PR to see the breakdown, no strong opinion)

MelSumner commented:
@amyrlam I don't have strong opinions on it- I commit and push frequently for...idk, no reason that is super apparent to me now that I think about it. Happy to do whatever makes sense, I don't usually review per commit but rather by overall changes. Totally can adjust.

amyrlam commented:
@MelSumner cool, yeah not trying to derail the PR, just a suggestion, to make it easier / faster for the reviewer. i'll take a stab at some guidelines separately and share so we can start to flesh out our "process" as a team

maybe we can both pair code review in the future, each reviewing something aloud that we haven't previously paired on?

MelSumner commented:
Sure sounds good!

MelSumner commented:
Update: 

- addressed PR review items
- addressed missing items in the dummy app that were throwing errors in the console
- rebased to be current

Would like to get this merged (squash/merge is fine) and then make other improvements separately.

Dhaulagiri commented:
:shipit:  I say we land this and followup on any details
